### DISCUSSION TITLE
redirect on getOne

### DISCUSSION THREAD

**Discussion by dankeyy** [pocketbase/pocketbase#3238] [Q&A]
So i have a collection with record, each record has it's uploader field.
i want that if a user is asking to getOne record that isn't his, an operation of copying relevant record entries will start (into a new record with him as the uploader), and i want him to be redirected to that new record.
Mostly what im interested in is the redirect, like where do i make this hook

essentially, i could've probably made a hook to view to initiate that copying, then with a relation field ask on the client side for the new copied record instead
but this seems suboptimal, would like to hear your opinions on it

**Comment by ganigeorgiev** [comment]
I don't understand the use case of this.

Why do you need a redirect?
Wouldn't returning a custom record response suffice?
How would you identify whether an existing record already has a "duplicate" in case the user tries to retrieve second time with the id of the record that they don't own?

In general you can create the duplicated record in the [OnRecordViewRequest](https://pocketbase.io/docs/go-event-hooks/#onrecordviewrequest) and return a custom response with the new record using something like `e.HttpContext.JSON(200, newRecord)`.
